mongoose.c

```diff
@@ -1060,7 +1060,7 @@ static void ns_handle_udp(struct ns_connection *ls) {
 }
 
 static void ns_add_to_set(sock_t sock, fd_set *set, sock_t *max_fd) {
-  if (sock != INVALID_SOCKET) {
+  if ( (sock != INVALID_SOCKET) && (sock >= 0) && (sock < FD_SETSIZE) ) {
```

Comment:
Visual C++ V547 Expression 'sock >= 0' is always true. Unsigned type value is always >= 0